### PR TITLE
feat(docs): introduce static docs built with vitepress

### PR DESCRIPTION
**Pull request by lucpotage**
Hi @antfu,

When moving from Windi CSS to UnoCSS, I found it difficult to understand the on-demand concept of UnoCSS and how to get an ISO experience from Windi CSS. While interactive docs are great to check that utilities work, it does not help finding the right information about presets, but also transformers that I wrongly thought were provided by default. There are articles with the guide prefix, but I found traditional static docs easier to browse: no need to know to find out.

Since the Nuxt team provided a great starter template for docs with Docus. I spent some time copying content from individual files to a new docs directory that is just waiting to be published (I suppose some CD setup would be required).

- [x] Install Docus boilerplate
- [x] Add content from individual README files
- [x] Adapt content with Docus components for alerts, embedded playgrounds, etc
- [x] Create a landing page with pointers to Getting started and Guide entries, and sponsors
- [x] Add favicon and cover image for social networks
- [x] Add license and contributors info in the footer, and links to Nuxt and Netlify
- [ ] Configure Netlify
- [ ] Check that links to Interactive docs and Playground work from the new docs

I obviously kept the original README files since they are useful as descriptions of individual packages in npm registry. I’m sure we could find a way to avoid duplicating content between the new docs and those README files, but I’m not sure how, especially since some files are not exactly the same (few Docus components).

I hope you’ll enjoy this new docs! 😃

### CODE REVIEW

**Comment by antfu** [comment]
Hey, thanks for working on it, and I am happy to have it, it can be a good start to introduce more explanation of the concepts.

I have a few feedback before we move on:

- I prefer to use VitePress over Docus, to align more with Vite ecosystem
- We could make the README of each package minimal and link to the corresponding page (you can assume the docs site is deployed under `https://unocss.dev`

Once you are done, I will merge it and handle the rest deploy setup. Thanks

**Comment by lucpotage** [comment]
Thanks for your super fast feedback. Here is the Vitepress version!

I slightly changed the routes to make them match the package names to have something like `unocss.dev/preset-mini` for instance. I also double checked dead links.

I kept the default theme, feel free to tell me if you have any special color palette in mind.

**Comment by userquin** [comment]
@lucpotage add `docs` to `pnpm-workspace.yml` file:
 ```yml
packages:
  - bench
  - docs
  - playground
  - interactive
  - 'packages/*'
  - 'test/fixtures/*'
  - examples/vue-cli4
  - examples/vue-cli5
```

**Comment by userquin** [comment]
`docs/src/public` folder should be on docs root folder: move public folder from here `docs/src/public` to here `docs/public`

**Comment by userquin** [comment]
@antfu I can include team cards, contributors, sponsors and so on... I can include previous stuff in this PR or wait an initial version merged.

File: docs/.vitepress/config.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
+import { defineConfig } from 'vitepress'
+
+const nav = [
+  { text: 'Getting started', link: '/presets' },
+  { text: 'Guide', link: 'preset-uno' },
+  { text: 'Reference', link: '/interactive' },
+  { text: 'Playground', link: '/play' },
```
Comment (userquin) [inline docs/.vitepress/config.ts:7]: Reference and playground should be external links to the corresponding sites

Comment (lucpotage) [reply docs/.vitepress/config.ts:7]: I was about to add the following links but @antfu mentioned the `unocss.dev` domain that I didn’t know was available.

``` js
{ text: 'Reference', link: 'https://uno.antfu.me/reference/' },
{ text: 'Playground', link: 'https://uno.antfu.me/play/' },
```

Regarding the interactive docs, what’s the preferred route? `interactive` or `reference`?

**Comment by lucpotage** [comment]
> `docs/src/public` folder should be on docs root folder: move public folder from here `docs/src/public` to here `docs/public`

I actually tried this at the beginning but it does not work. So I looked at available examples in GitHub. StackBlitz Docs for instance have a `public` dir inside `docs` which is their `srcDir` in `config.ts`. Did I miss something?

File: docs/.vitepress/config.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  description: 'The instant on-demand Atomic CSS engine',
+  head: [
+    ['meta', { name: 'og:title', content: 'UnoCSS' }],
+    ['meta', { property: 'og:image', content: '/cover/default.png' }],
+  ],
+  lang: 'en-US',
+  srcDir: './src',
+  cleanUrls: true,
+  themeConfig: {
+    logo: '/logo.svg',
```
Comment (userquin) [inline docs/.vitepress/config.ts:107]: add suggest changes here: 
```ts
    editLink: {
      pattern: 'https://github.com/unocss/unocss/docs/edit/main/docs/:path',
      text: 'Suggest changes to this page',
    },
```

File: docs/.vitepress/config.ts (hunk 1 above)
Comment (userquin) [reply docs/.vitepress/config.ts:7]: The domain is for the new docs deployed on netlify, the links to include are:
- Interactive Docs (should change Reference): https://uno.antfu.me/
- Playground: https://uno.antfu.me/play/

File: docs/.vitepress/config.ts (hunk 2 above)
Comment (lucpotage) [reply docs/.vitepress/config.ts:107]: Done: https://github.com/unocss/unocss/pull/2386/commits/5dc4664098c7240cf25d36bd9e49d1779b2b98c7

**Comment by userquin** [comment]
> I actually tried this at the beginning but it does not work. So I looked at available examples in GitHub. StackBlitz Docs for instance have a `public` dir inside `docs` which is their `srcDir` in `config.ts`. Did I miss something?

I need to check latest changes in VitePress.

You can test it removing the `srcDir` option and moving all inside src folder to docs root. I didn't see that flag, for example, on VitePWA docs repo we have md files at root, or in Vitest repo (docs folder) we have md files on docs root folder.

File: docs/.vitepress/config.ts (hunk 1 above)
Comment (lucpotage) [reply docs/.vitepress/config.ts]: Thanks. Done: https://github.com/unocss/unocss/pull/2386/commits/40908aaf86d27027be2571b912a6aa68a963fe46

Comment (antfu) [reply docs/.vitepress/config.ts]: I will move them to the new domain, no worries

Comment (userquin) [reply docs/.vitepress/config.ts]: can you replace `Reference` with `Interactive Docs`?

Comment (lucpotage) [reply docs/.vitepress/config.ts]: Sure! https://github.com/unocss/unocss/pull/2386/commits/943cb492f7c57a98540d26a0f1d935c346f94ce4

File: docs/.vitepress/config.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  title: 'UnoCSS',
+  titleTemplate: 'UnoCSS',
+  description: 'The instant on-demand Atomic CSS engine',
+  head: [
+    ['meta', { name: 'og:title', content: 'UnoCSS' }],
+    ['meta', { property: 'og:image', content: '/cover/default.png' }],
+  ],
+  lang: 'en-US',
+  srcDir: './src',
+  cleanUrls: true,
```
Comment (userquin) [inline docs/.vitepress/config.ts:105]: add `  lastUpdated: true,`

Comment (lucpotage) [reply docs/.vitepress/config.ts:106]: Sure: https://github.com/unocss/unocss/pull/2386/commits/21ceef82cac60b8869d1bf780870a5e72d309353

**Comment by userquin** [comment]
I'm fixing types in config.ts module, do you want I move all src one level up?

**Comment by lucpotage** [comment]
Does it mean you prefer having the md files at the root level?

**Comment by userquin** [comment]
> Does it mean you prefer having the md files at the root level?

No, we want the md files on docs root level not inside src folder.

**Comment by antfu** [comment]
@lucpotage thanks for working on it! I just restructured the document a bit. It would be better to pull first before making changes in case of conflictions.

**Comment by lucpotage** [comment]
Thanks @antfu. Sorry for the mess. I use the UI Sync feature of VSCode that says it does pull before push. 😳

**Comment by antfu** [comment]
I have merged this PR into `/docs-next` branch as there are still a lot of things to do before making it public. @lucpotage if you want to help, feel free to send more PRs against that branch. Thanks

**Pull request merged**